tsconfig.json

```diff
@@ -12,10 +12,17 @@
         "noImplicitThis": true,
         "noImplicitReturns": true,
         "strictNullChecks": true,
-        "noUnusedParameters": true
+        "noUnusedParameters": true,
+        "baseUrl": "./",
+        "paths": {
+            "*": [
+                "node_modules/vscode/*",
```

Comment:
Why adding this path?  Related to vscode.d.ts issue?

Reply:
Ah, okay , this is to fix the vscode.d.ts errors when using npm link.

@PrashanthCorp @nturinski @EricJizbaMSFT FYI:
Personally I prefer using npm pack in the UI package, then npm install <path-to-local-tgz> rather than using npm link because:
1) if you do "git clean -xdf" in your client project, it will follow the hard link and completely wipe out your linked sources with no way to recover changes
2) some things will work with npm link but will cause build errors when you actually publish. npm pack gives you exactly the same situation as publishing globally.
3) you don't get the vscode.d.ts issue.

Reply:
Yes see this PR for more info: https://github.com/Microsoft/vscode-azurefunctions/pull/531
I can pull this out into a separate PR if you want - I was just being lazy.

Anyways I think it's just a personal preference on whether people use `npm link` vs `npm pack` (I haven't run into your issues with 1 and 2). This just makes it easier for those that prefer `npm link` (aka me)

Reply:
Fair enough.  I've run into 1 and 2. :-(